### ISSUE TITLE
Syntax Error while versioning iOS

### ISSUE THREAD

**Issue by aruldd** [stovmascript/react-native-version#52]
> PS F:\Shared Desktop\ROBOCRAZE\RnD\alphacare\hospital_app_react> yarn version --minor
yarn version v1.6.0
warning ..\package.json: No license field
info Current version: 0.1.0-beta
question New version:
error Invalid semver version
question New version: 0.2.0
info New version: 0.2.0
$ react-native-version
[RNV] Versioning Android...
[RNV] Android updated
[RNV] Versioning iOS...
[RNV] SyntaxError: Expected "\"", "\\'", "\\\"", "\\n", or [^\\"] but "\\" found.

After setting up post version hook, running the command `yarn version --minor` produces this error.

Platform: Windows 10

**Comment by stovmascript** [comment]
Looks similar to: https://github.com/stovmascript/react-native-version/issues/38#issuecomment-372974523 and: https://github.com/stovmascript/react-native-version/issues/38#issuecomment-385656056

**Comment by stovmascript** [comment]
@aruldd Is it possible you have manually set up some paths in your Xcode project file?

**Comment by VinceBT** [comment]
Found the issue, the problem comes from React Native made in Windows that will have backslashed paths inside pbxproj. It's not a problem from your package at all @stovmascript .
Check strings that are like this:
```
"\"$(SRCROOT)/$(TARGET_NAME)\"",
or
"$(SRCROOT)\..\node_modules\react-xxxx",
```
and change them to 
```
"$(SRCROOT)/$(TARGET_NAME)",
accordingly
"$(SRCROOT)/../node_modules/react-xxxx",
```

**Comment by stovmascript** [comment]
@VinceBT I also suspect that it's more of an issue with `react-native link` because a newly initialized project doesn't have any `$(SRCROOT)` strings yet. Which you explain why everything worked for me in: https://github.com/stovmascript/react-native-version/issues/38#issuecomment-383421578.

**Comment by VinceBT** [comment]
It's just because Windows uses \ instead of / for paths and when paths are written to pbxproj, they will contain backslashes, it still works when your compile on a mac but it will crash with the pbxproj-dom parser

**Comment by VinceBT** [comment]
@aruldd Tell us if it worked for you so @stovmascript can close

**Comment by Maushundb** [comment]
I'm getting this issue when running on Mac OSX, is that expected? I'm on 3.2.0

**Comment by Maushundb** [comment]
Yeah there seems to be some paths in pbxproj that cocoapods installs that has the windows syntax as well. Had to go in and edit them manually, now it works. This does happen on macs as well apparently, might wanna update docs.

**Comment by stovmascript** [comment]
Hmm, this seems like a bug with cocoapods, or even some specific packages. I'm wondering if we could find the offending upstream code and report it...

**Comment by stovmascript** [comment]
Or we could patch `pbxproj-dom` or find an alternative.

**Comment by Maushundb** [comment]
If it helps, offending line was:

```
shellScript = "export SENTRY_PROPERTIES=sentry.properties\nexport NODE_BINARY=node\n../node_modules/@sentry/cli/bin/sentry-cli react-native xcode \\\n../node_modules/react-native/scripts/react-native-xcode.sh\n";
```

Had to remove the `\\` from `\\\n`

from https://www.npmjs.com/package/@sentry/react-native

**Comment by vladwhd** [comment]
I have a more "funny" case. My error was "SyntaxError: Expected "\"", "\\'", "\\\"", "\\n", or [^\\"] but "\\" found.", so I searched for \\ in the project.pbxproj and the only place where it was present was `			shellScript = "# Name of the resource we're selectively copying\nFILENAME=GoogleService-Info.plist\n\nBUILD_TYPE=$(echo ${CONFIGURATION} | sed -E \"s/\\s*(Release|Debug) \\s*//\")\nORIGINAL_PATH=${PROJECT_DIR}/${TARGET_NAME}/Firebase/${BUILD_TYPE}/${FILENAME}\nDESTINATION_DIR=${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app\n\n# Make sure the prod version of GoogleService-Info.plist exists\necho \"Looking for ${ORIGINAL_PATH}\"\nif [ ! -f $ORIGINAL_PATH ]\nthen\n    echo \"No ${FILENAME} found for build type ${BUILD_TYPE}. Please ensure it's in the proper directory.\"\n    exit 1\nfi\n\necho \"Will copy ${ORIGINAL_PATH} to final destination: ${DESTINATION_DIR}\"\n\ncp \"${ORIGINAL_PATH}\" \"${DESTINATION_DIR}\"\n";
`

So basically it failed parsing the shell script, which is kind of weird. I removed this script and the error was gone, so it's this indeed.
I will change my script to fix this, but it shouldn't have happened in the first place.

**Comment by fabricio0915** [comment]
> If it helps, offending line was:
> 
> ```
> shellScript = "export SENTRY_PROPERTIES=sentry.properties\nexport NODE_BINARY=node\n../node_modules/@sentry/cli/bin/sentry-cli react-native xcode \\\n../node_modules/react-native/scripts/react-native-xcode.sh\n";
> ```
> 
> Had to remove the `\\` from `\\\n`
> 
> from https://www.npmjs.com/package/@sentry/react-native

your command line appers like this? 

```
shellScript = "export SENTRY_PROPERTIES=sentry.properties\nexport NODE_BINARY=node\n../node_modules/@sentry/cli/bin/sentry-cli react-native xcode \n../node_modules/react-native/scripts/react-native-xcode.sh\n";
```

**Comment by sanjaypojo** [comment]
Facing the same problem after adding a custom build script, have been using the library without any issues for a year, but stuck now. Lots of backslashes in the script to escape quotes and spaces, so not sure how to fix it.

**Comment by sanjaypojo** [comment]
Just to clarify though -- I'm facing this issue on macOS. Here's the exact error:

`[RNV] SyntaxError: Expected ",", "/*", or [ \t\n\r] but "B" found.`

**Comment by otkach-yara** [comment]
I have fixed this issue by moving the script, which caused the error, from `project.pbxproj` to separate `.sh` file and then just added a reference to that file. Also you should make your `sh` file executable by `chmod +x script.sh`

<img width="697" alt="Screenshot 2021-05-18 at 17 53 52" src="https://user-images.githubusercontent.com/64953452/118673888-0b106000-b802-11eb-9b25-8c6cf7179776.png">

**Comment by bimusiek** [comment]
In our case it was newest react with @sentry/react-native having offending lines in shell script ([copied from official Sentry documentation about workaround for RN >0.69](https://docs.sentry.io/platforms/react-native/troubleshooting/#react-native-069-and-higher))
Here is the working shell script line in case anyone needs it:
```
/bin/sh -c "$WITH_ENVIRONMENT '$SENTRY_CLI_PATH react-native xcode $REACT_NATIVE_XCODE'"
```

**Comment by marcinniemirski** [comment]
Here is the final version of working shellScript in Sentry v5 and RN v0.71 

```shell
shellScript = "set -e\nexport SENTRY_PROPERTIES=sentry.properties\nexport EXTRA_PACKAGER_ARGS=\"--sourcemap-output $DERIVED_FILE_DIR/main.jsbundle.map\"\nWITH_ENVIRONMENT=\"../node_modules/react-native/scripts/xcode/with-environment.sh\"\nREACT_NATIVE_XCODE=\"../node_modules/react-native/scripts/react-native-xcode.sh\"\nSENTRY_CLI_PATH=\"../node_modules/@sentry/cli/bin/sentry-cli\"\n/bin/sh -c \"$WITH_ENVIRONMENT $SENTRY_CLI_PATH react-native xcode\n$REACT_NATIVE_XCODE\"\n/bin/sh ../node_modules/@sentry/react-native/scripts/collect-modules.sh\n";
```

**Comment by n-ii-ma** [comment]
I'm experiencing this issue with Sentry.

I suspect this line is causing the problem, but I don't know how to fix it:

`shellScript = "export SENTRY_PROPERTIES=sentry.properties\nexport EXTRA_PACKAGER_ARGS=\"--sourcemap-output $DERIVED_FILE_DIR/main.jsbundle.map\"\nset -e\n\nWITH_ENVIRONMENT=\"../node_modules/react-native/scripts/xcode/with-environment.sh\"\nREACT_NATIVE_XCODE=\"../node_modules/react-native/scripts/react-native-xcode.sh\"\n\n/bin/sh -c \"$WITH_ENVIRONMENT \\\"../node_modules/@sentry/cli/bin/sentry-cli react-native xcode $REACT_NATIVE_XCODE\\\"\"\n\n/bin/sh ../node_modules/@sentry/react-native/scripts/collect-modules.sh\n";
`

Any way to solve this?

**Comment by tconroy** [comment]
Hey @n-ii-ma  - also running into this in 2023. Curious if you found a fix?

**Comment by alainib** [comment]
i have the same error on MacOs now. any one fixed this please ?

**Comment by erickgtzh** [comment]
> i have the same error on MacOs now. any one fixed this please ?

were you able to fix it?

**Comment by 29er** [comment]
SyntaxError: Expected "\"", "\\'", "\\\"", "\\n", or [^\\"] but "\\" found.

similar to above it was Sentry shell script. I am on a mac.
shellScript = "export SENTRY_PROPERTIES=sentry.properties\nexport ....."

in several places I had to replace `$WITH_ENVIRONMENT \\\` with `$WITH_ENVIRONMENT \ `  and  `$REACT_NATIVE_XCODE\\\`  with `$REACT_NATIVE_XCODE\`

**Comment by Leslie-Wong-H** [comment]
Faced the same issue with Mac. Resolved with changing 
```
shellScript = "# This script configures Expo modules and generates the modules provider file.\nbash -l -c \"./Pods/Target\\ Support\\ Files/Pods-tracemoe_react_native/expo-configure-project.sh\"\n";
```
to 
```
shellScript = "# This script configures Expo modules and generates the modules provider file.\nbash -l -c \"./Pods/Target' 'Support' 'Files/Pods-tracemoe_react_native/expo-configure-project.sh\"\n";
```
The key: `\\ ` to `' '`
Root reason - the file path with space in it.

**Comment by moisesmiss** [comment]
> Faced the same issue with Mac. Resolved with changing
> 
> ```
> shellScript = "# This script configures Expo modules and generates the modules provider file.\nbash -l -c \"./Pods/Target\\ Support\\ Files/Pods-tracemoe_react_native/expo-configure-project.sh\"\n";
> ```
> 
> to
> 
> ```
> shellScript = "# This script configures Expo modules and generates the modules provider file.\nbash -l -c \"./Pods/Target' 'Support' 'Files/Pods-tracemoe_react_native/expo-configure-project.sh\"\n";
> ```
> 
> The key: `\\ ` to `' '` Root reason - the file path with space in it.

This works form me, thanks!
